File: 16456 - 하와아 대학생쨩 하와이로 가는 거시와요/16456.py
import sys


def solution(n):
    arr = [0] * (n + 3)
    arr[0] = 1
    arr[1] = 1
    arr[2] = 2
    for i in range(3, n):
        arr[i] = (arr[i - 1] + arr[i - 3]) % 1000000009
    return arr[n - 1]


n = int(sys.stdin.readline())
print(solution(n) % 1000000009)

File: 16456 - 하와아 대학생쨩 하와이로 가는 거시와요/16456.cs
using System;


namespace _16456
{
    class Program
    {
        static void Main(string[] args)
        {
            int mod = 1000000009;
            int n = int.Parse(Console.ReadLine());
            int[] arr = new int[n + 3];
            arr[0] = 1;
            arr[1] = 1;
            arr[2] = 2;
            for (int i = 3; i < n; i++)
            {
                arr[i] = (arr[i - 1] + arr[i - 3]) % mod;
            }
            Console.WriteLine((arr[n - 1] % mod).ToString());
        }
    }
}
